transformer_engine/pytorch/permutation.py

```diff
@@ -468,6 +495,7 @@ def moe_permute(
     num_out_tokens: int = -1,
     max_token_num: int = -1,
     map_type: str = "mask",
+    probs: torch.Tensor = None,
 ) -> Tuple[torch.Tensor, torch.Tensor]:
```

Comment:
We should make sure the type hints are accurate:
```suggestion
) -> Union[Tuple[torch.Tensor, torch.Tensor],Tuple[torch.Tensor, torch.Tensor, torch.Tensor]]:
```
Messy type hints are a clue that the design is messy. I prefer to avoid variadic returns like this since users might need to have extra logic to unpack the outputs. Better to always return a tuple with 3 elements, where `permuted_probs` is `None` if not needed:
```python
) -> Tuple[torch.Tensor, torch.Tensor, Optional[torch.Tensor]]:
```

Reply:
Hi, Tim,

I'm also not sure how to handle this part. If we modify the return numbers, we may break some compatibility. The current MCore assumes that this function returns two tensors.

What about adding two new functions `moe_permute_with_probs` and `moe_sort_chunks_by_index_with_probs` and leaving the previous function unchanged? Do you think it is a good idea?

Reply:
I added new functions `moe_permute_with_probs` and `moe_sort_chunks_by_index_with_probs` and kept the original ones unchanged.